dynawo/sources/Simulation/DYNSimulation.cpp

```diff
@@ -877,20 +877,32 @@ Simulation::simulate() {
   } catch (const Terminate& t) {
     Trace::warn() << t.what() << Trace::endline;
     model_->printMessages();
-    if (timetableOutputFile_ != "")
-        remove(timetableOutputFile_);
+    endSimulationWithError(criteriaChecked);
   } catch (const Error& e) {
     Trace::error() << e.what() << Trace::endline;
-    if (timetableOutputFile_ != "")
-        remove(timetableOutputFile_);
+    endSimulationWithError(criteriaChecked);
     throw;
```

Comment:
Out of curiosity, in which case do we end up through the terminate part and in which case we go through the error part? I don't have any idea on what is behind the terminate thing (is it a terminate signal given by the user?)

Reply:
this is to handle the terminate keyword of modelica